app/hallebarde/infrastructure/exchange_repository.py

```diff
@@ -49,6 +54,24 @@ def get_by_download_token(download_token: str) -> Optional[Exchange]:
     return _map_exchange_from_item(items[0]) if items else None
 
 
+def get_before_time(creation_time: datetime) -> List[Exchange]:
+    timestamp = Decimal(creation_time.timestamp())
+    table = _get_dynamodb_table()
+
+    scan_params = {'FilterExpression': Attr('creation_time').lte(timestamp),
+                   'Limit': 50}
```

Comment:
OK, this was a pretty tough one diving into dynamodb specs.

General rule is that dynamodb (through boto) may paginate the results of your scan. My intent was to "force" pagination by setting a low limit for maximum number of results, in order to avoid too long tests that would have to load thousands of items and then querying thousands of items

But I had mixed up everything in my mind: 
- There is a limit of 1 MB for maximum query size
 - We reach this limit approximately at 5000 elements (meaning we would delete 5000 exchanges at a time)
- There is a limit of 100 maximum result, but this is only for `batch_get_item` method, which we do not use

So I will remove all this pagination stuff for the moment, and we may wait to have actual users for managing all this. I think we should take a time at a moment in order to assess DynamoDB relevance for our cases, maybe once we will have stabilized the data model?

Because today we do lot of scanning without indexes, and though we may not hit walls right now, we should not totally close our eyes on this matter. What do you think @Mehdi-H ?

Reply:
Ok, I get it. But, this is a unit test with just a few object in the db. I think we shouldn't mix things up between functional validation and performance issues that we don't have for now: I don't know how many elements we will have in the db at a given time.

For now, if every of our french users in our company puts 1 file on hallebarde (and I don't think that everyone will put 1 file there during the same week), we'll have like ~1000 elements max.

I think we can let performance and monitoring issues for another "git issue".

Reply:
I do agree. I will remove these pagination stuff for the moment.